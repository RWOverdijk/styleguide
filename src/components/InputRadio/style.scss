/** @define InputRadio */

@import "../../css/variables";
@import "../../css/mixins";

.Van-InputRadio {
  position: relative;
  width: 20px;
  height: 20px;
  display: inline-block;

  &-input {
    position: absolute;
    top: 0;
    left: 0;
    width: 20px;
    height: 20px;
    margin: 0;
    opacity: 0;
    z-index: 10;

    & + .Van-InputRadio-styled {
      /* postcss-bem-linter: ignore */
      &::before {
        content: '';
        position: absolute;
        top: 0;
        left: 0;
        width: 20px;
        height: 20px;
        background-color: $color-white;
        border: 1px solid $color-grey-light;
        border-radius: 50%;
      }
    }

    @keyframes scale {
      from {
        transform: scale(0);
      }

      to {
        transform: scale(1);
      }
    }

    &:checked + .Van-InputRadio-styled {
      /* postcss-bem-linter: ignore */
      &::after {
        content: '';
        position: absolute;
        top: 4px;
        left: 4px;
        width: 12px;
        height: 12px;
        background-color: $color-pink-primary;
        border-radius: 50%;
        animation-duration: 0.2s;
        animation-name: scale;
      }

      /* postcss-bem-linter: ignore */
      &::before {
        content: '';
        position: absolute;
        top: 0;
        left: 0;
        width: 20px;
        height: 20px;
        background-color: $color-white;
        border: 1px solid $color-pink-primary;
        border-radius: 50%;
      }
    }
  }
}
