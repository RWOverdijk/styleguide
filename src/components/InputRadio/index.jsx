import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import './style.scss';

class InputRadio extends PureComponent {
  render() {
    const { meta, input, className, ...props } = this.props;

    return (
      <span className={classnames('Van-InputRadio', className)}>
        <input type="radio" className="Van-InputRadio-input" {...props} {...input} />
        <span className="Van-InputRadio-styled" />
      </span>
    );
  }
}

InputRadio.defaultProps = {
  className: '',
  meta: {},
  input: {}
};

InputRadio.propTypes = {
  className: PropTypes.string,
  meta: PropTypes.object,
  input: PropTypes.object
};

export default InputRadio;
