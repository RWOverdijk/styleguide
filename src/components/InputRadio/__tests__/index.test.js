import React from 'react';
import renderer from 'react-test-renderer';
import InputRadio from '../';

describe("<InputRadio>", () => {
  it("renders the correct markup", () => {
    const tree = renderer.create(
      <InputRadio />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific className", () => {
    const tree = renderer.create(
      <InputRadio className="my-class" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific attributes", () => {
    const tree = renderer.create(
      <InputRadio value="on" data-id="0" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
