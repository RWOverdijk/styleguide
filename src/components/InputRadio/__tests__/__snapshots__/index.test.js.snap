// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputRadio> renders the correct markup 1`] = `
<span
  className="Van-InputRadio"
>
  <input
    className="Van-InputRadio-input"
    type="radio"
  />
  <span
    className="Van-InputRadio-styled"
  />
</span>
`;

exports[`<InputRadio> renders with specific attributes 1`] = `
<span
  className="Van-InputRadio"
>
  <input
    className="Van-InputRadio-input"
    data-id="0"
    type="radio"
    value="on"
  />
  <span
    className="Van-InputRadio-styled"
  />
</span>
`;

exports[`<InputRadio> renders with specific className 1`] = `
<span
  className="Van-InputRadio my-class"
>
  <input
    className="Van-InputRadio-input"
    type="radio"
  />
  <span
    className="Van-InputRadio-styled"
  />
</span>
`;
